### PR TITLE
RUST-1525 Update AWS tests to use new scripts

### PR DESCRIPTION
**Pull request by isabelatkinson**
Removes all uses of the legacy shell by migrating to the new AWS scripts in drivers-evergreen-tools. Also completes RUST-1777 by upgrading to Ubuntu 20.04 (the new arn is no longer relevant now that we're using the secrets manager).

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -79,9 +79,6 @@ buildvariants:
     run_on:
       - ubuntu2204-arm64-small
     expansions:
-      # Needed to skip installation of the legacy shell in boostrap mongo-orchestration, which is
-      # not available on Ubuntu 22.04.
-      SKIP_LEGACY_SHELL: true
```
Comment (isabelatkinson) [inline .evergreen/config.yml:84]: This variable is no longer respected by drivers-evergreen-tools so I removed our uses of it

File: .evergreen/config.yml [hunk 2]
```diff
@@ -266,13 +257,10 @@ buildvariants:
 
   - name: aws-auth
     display_name: "AWS Authentication"
-    patchable: false
+    # patchable: false
```
Comment (isabelatkinson) [inline .evergreen/config.yml:260]: I'll uncomment this before merging

File: .evergreen/config.yml [hunk 3]
```diff
@@ -266,13 +257,10 @@ buildvariants:
...
     run_on:
-      # Use ubuntu1804. AWS tasks require legacy `mongo` shell. Legacy `mongo` shell is not available on newer distros.
-      - ubuntu1804-small
+      - ubuntu2004-small
```
Comment (isabelatkinson) [inline .evergreen/config.yml:262]: Using an older Ubuntu version here because the ECS test doesn't work on 22.04

**Review by abr-egn — APPROVED** [review]
LGTM!

File: .evergreen/env.sh [hunk 4]
```diff
@@ -9,6 +9,7 @@ export NVM_DIR="${NODE_ARTIFACTS_PATH}/nvm"
 
 . ${CARGO_HOME}/env
 
+set +u # allow undefined variable for $OS
 if [[ "$OS" == "Windows_NT" ]]; then
```
Comment (kevinAlbs) [inline .evergreen/env.sh:13]:
```suggestion
if [[ "$OSTYPE" == "cygwin" ]]; then
```

Suggest using the Bash variable `OSTYPE`. It is a variable set by Bash: https://www.gnu.org/software/bash/manual/html_node/Bash-Variables.html and this script defines bash in the shebang.

`OSTYPE` is also used in the C driver scripts. Example: https://github.com/mongodb/mongo-c-driver/blob/5e5784d814b2e3912c2855ac2b9a172267b8235f/.evergreen/scripts/add-build-dirs-to-paths.sh#L3-L4

File: .evergreen/config.yml [hunk 5]
```diff
@@ -1476,46 +1370,22 @@ functions:
           cp $PROJECT_DIRECTORY/.evergreen/run-mongodb-aws-ecs-test.sh $ECS_SRC_DIR/.evergreen
```
Comment (kevinAlbs) [inline .evergreen/config.yml:1370]: Suggest removing the `sed` line above. I expect it is no longer needed. Adding `TestData = {};\n` does not appear in the [C driver config](https://github.com/mongodb/mongo-c-driver/blob/5e5784d814b2e3912c2855ac2b9a172267b8235f/.evergreen/scripts/run-aws-tests.sh#L116-L133).

**Review by kevinAlbs — APPROVED** [review]
LGTM. I left two minor suggestions.

File: .evergreen/config.yml (hunk 5 above)
Comment (isabelatkinson) [reply .evergreen/config.yml:1367]: updated

File: .evergreen/env.sh (hunk 4 above)
Comment (isabelatkinson) [reply .evergreen/env.sh]: updated

**Pull request merged**